src/proof/supernova.rs

```diff
@@ -236,30 +243,37 @@ where
 
 /// A struct for the Nova prover that operates on field elements of type `F`.
 #[derive(Debug)]
-pub struct SuperNovaProver<F: CurveCycleEquipped, C: Coprocessor<F>> {
+pub struct SuperNovaProver<
+    'a,
+    F: CurveCycleEquipped,
+    C: Coprocessor<F> + 'a,
+    M: MultiFrameTrait<'a, F, C>,
+> {
     // `reduction_count` specifies the number of small-step reductions are performed in each recursive step of the primary Lurk circuit.
     reduction_count: usize,
     lang: Lang<F, C>,
+    _phantom: PhantomData<&'a M>,
 }
 
-impl<'a, F: CurveCycleEquipped, C: Coprocessor<F>> PublicParameters for PublicParams<'a, F, C>
+impl<F: CurveCycleEquipped, C1: StepCircuit<F>> PublicParameters for PublicParams<F, C1>
 where
     <<G1<F> as Group>::Scalar as ff::PrimeField>::Repr: Abomonation,
     <<G2<F> as Group>::Scalar as ff::PrimeField>::Repr: Abomonation,
 {
 }
```

Comment:
`PublicParameters` is a marker trait used only to put a bound on the associated type of `Prover`. I think we can remove it and its three instances: the fact that the associated type is called `PublicParameters`, and that each of the three structs chosen for the three instances of `Prover ` are called `PublicParams` should be enough to convey the meaning. And the `crate::proof` module could use some simplification.